File: tools/dbench/answers/go_haiku/s2/e1_parse_fallback.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	strs := []string{"12", "x", "7", "-", "30"}
	sum := 0
	bad := 0

	for _, s := range strs {
		num, err := strconv.Atoi(s)
		if err != nil {
			bad++
		} else {
			sum += num
		}
	}

	fmt.Println(sum)
	fmt.Println(bad)
}

File: tools/dbench/answers/rust_sonnet/s2/e1_parse_fallback.rs
fn main() {
    let values = ["12", "x", "7", "-", "30"];
    let mut sum = 0i64;
    let mut bad = 0usize;
    for v in &values {
        match v.parse::<i64>() {
            Ok(n) => sum += n,
            Err(_) => bad += 1,
        }
    }
    println!("{}", sum);
    println!("{}", bad);
}
